### ISSUE TITLE
RASPIGNSS NV08C no glonass fix

### ISSUE THREAD

**Issue by zuliani71** [tomojitakasu/RTKLIB#91]
Hi, I've a baseline between a VRS from a caster and the NV08C device (raspignss+raspberry with rtklib installed on it). I can get a fix but just for GPS and not for GLONASS. Running obs from the rtkrcv console gives me:

SAT C1 L1 L2  Fix1  Fix2  Sl1  Sl2  Lock1  Lock2 Rj1 Rj2  P1Res P2Res  L1Res  L2Res    Az   El
G02 OK OK  -   FIX     -    0    0   1527      0   0   0 -0.517 0.000 0.0031 0.0000  43.4 14.8
G05 OK OK  -   FIX     -    4    0   1573      0   0   0  1.280 0.000 0.0016 0.0000  76.0 17.3
G12 OK OK  -   FIX     -    4    0   1573      0   0   0  0.312 0.000 0.0034 0.0000 117.2 14.2
G20 OK OK  -   FIX     -    0    0   1511      0   0   0  0.717 0.000 0.0013 0.0000  92.7 19.8
G21 OK OK  -   FIX     -    4    0   1573      0   0   0 -0.030 0.000 0.0028 0.0000 184.2 43.4
G25 OK OK  -   FIX     -    4    0   1149      0   0   0  0.487 0.000 0.0023 0.0000 116.4 51.7
G26 OK OK  -   FIX     -    0    0   1527      0   0   0  0.457 0.000 0.0034 0.0000 302.5 29.9
G29 OK OK  -   FIX     -    4    0   1573      0   0   0  0.000 0.000 0.0000 0.0000  42.2 71.1
G31 OK OK  -   FIX     -    4    0   1573      0   0   0 -0.302 0.000-0.0006 0.0000 264.8 52.6
R02  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000  32.6 11.4
R09  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000 277.6 73.3
R10  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000 327.2 25.8
R16  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000 169.0 41.5
R18  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000  40.4 40.3
R19  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000 273.0 70.8
R20  -  -  -     -     -    4    0      0      0   0   0  0.000 0.000 0.0000 0.0000 238.4 19.7

Is it any problem with GLONASS on rtklib and NV08C devices?

**Comment by astrodanco** [comment]
> I've a baseline between a VRS from a caster

 

Just as a sanity check, have you run single mode against the caster only as input to verify that it includes the GLONASS sats and that they at least get used in the single solution?

**Comment by zuliani71** [comment]
Ok I set the single mode. The obs command gives:

TIME(GPST) SAT R       P1(m)       P2(m)      L1(cyc)      L2(cyc)  D1(Hz)  D2(Hz) S1 S2 LLI
17:01:39.5 G05 1 24103113.35        0.00  -4311084.49         0.00  -817.2     0.0 37  0 0 0
17:01:39.5 G16 1 23786088.99        0.00  -8341526.71         0.00  3068.9     0.0 43  0 0 0
17:01:39.5 G20 1 23305307.57        0.00  -7442284.02         0.00   -68.2     0.0 45  0 0 0
17:01:39.5 G21 1 21539285.76        0.00 -19848631.02         0.00   917.5     0.0 48  0 0 0
17:01:39.5 G25 1 23154955.60        0.00  -8783779.75         0.00 -3313.2     0.0 45  0 0 0
17:01:39.5 G26 1 21631460.73        0.00 -19692877.01         0.00  2020.4     0.0 49  0 0 0
17:01:39.5 G29 1 21499622.21        0.00 -14178746.73         0.00 -2130.8     0.0 49  0 0 0
17:01:39.5 G31 1 22382547.89        0.00 -13927495.87         0.00 -2263.1     0.0 48  0 0 0
17:01:39.5 R02 1 24241292.15        0.00         0.00         0.00 -2067.6     0.0 25  0 0 0
17:01:39.5 R03 1 23086459.03        0.00  -6911835.17         0.00  1115.7     0.0 39  0 0 0
17:01:39.5 R04 1 24049564.17        0.00         0.00         0.00  3525.2     0.0 23  0 0 0
17:01:39.5 R09 1 20262810.20        0.00 -13763564.89         0.00 -2457.5     0.0 44  0 0 0
17:01:39.5 R10 1 20704022.34        0.00 -19155005.55         0.00  1892.0     0.0 48  0 0 0
17:01:39.5 R16 1 23434223.91        0.00    950021.15         0.00 -4772.8     0.0 41  0 0 0
17:01:39.5 R18 1 23119497.46        0.00  -1238551.62         0.00 -3616.2     0.0 39  0 0 0
17:01:39.5 R19 1 19862200.63        0.00 -17986132.91         0.00 -1089.1     0.0 49  0 0 0
17:01:39.5 R20 1 21334328.65        0.00 -16246714.58         0.00  2248.4     0.0 41  0 0 0
17:01:40.0 G05 2 23770424.68        0.00 124914474.74         0.00     0.0     0.0 42  0 0 0
17:01:40.0 G16 2 23453046.98        0.00 123246622.11         0.00     0.0     0.0 43  0 0 0
17:01:40.0 G20 2 22972541.09        0.00 120721582.23         0.00     0.0     0.0 41  0 0 0
17:01:40.0 G21 2 21206416.44        0.00 111440567.47         0.00     0.0     0.0 49  0 0 0
17:01:40.0 G25 2 22822482.29        0.00 119933016.30         0.00     0.0     0.0 46  0 0 0
17:01:40.0 G26 2 21298509.16        0.00 111924487.07         0.00     0.0     0.0 51  0 0 0
17:01:40.0 G29 2 21167051.26        0.00 111233759.98         0.00     0.0     0.0 49  0 0 0
17:01:40.0 G31 2 22049991.81        0.00 115873520.97         0.00     0.0     0.0 49  0 0 0
17:01:40.0 R02 2 23908658.16        0.00 127581193.48         0.00     0.0     0.0 34  0 0 0
17:01:40.0 R03 2 22753560.27        0.00 121801616.43         0.00     0.0     0.0 38  0 0 0
17:01:40.0 R09 2 19930237.97        0.00 106426401.92         0.00     0.0     0.0 44  0 0 0
17:01:40.0 R10 2 20371066.41        0.00 108589290.52         0.00     0.0     0.0 51  0 0 0
17:01:40.0 R16 2 23101858.05        0.00 123405970.36         0.00     0.0     0.0 38  0 0 0
17:01:40.0 R18 2 22787044.63        0.00 121638901.37         0.00     0.0     0.0 42  0 0 0
17:01:40.0 R19 2 19529512.89        0.00 104469895.74         0.00     0.0     0.0 49  0 0 0
17:01:40.0 R20 2 21001330.00        0.00 112303620.33         0.00     0.0     0.0 41  0 0 0

**Comment by zuliani71** [comment]
Finally I've found the reason of the missing GLONASS. To speed up FIX I've enabled:
pos1-sateph = brdc+sbas
which usually mades my FIX faster (the SBAAS solution is achivied very quick and so the rover position can be sent to the VRS MASTER which needs that info to stream corrections to the given rover). Unfortunately the brdc+sbas combination always prevents the GLONASS FIXing. Using:
pos1-sateph = brdc
finally gives me a correct FIX for both GPS and GLONASS
I don't know if that's a bug or if it's projected to act in that way.
